### DISCUSSION TITLE
8.7.0-pre.1

### DISCUSSION THREAD

**Discussion by wasabii** [ikvmnet/ikvm#426] [Announcements]
## What's Changed
* Return same Lambda where Lambda doesn't have captures by @TheLastRar in https://github.com/ikvmnet/ikvm/pull/376
* Enable .NET7 builds and tests by @wasabii in https://github.com/ikvmnet/ikvm/pull/378
* IKVM.Runtime cleanup by @wasabii in https://github.com/ikvmnet/ikvm/pull/383
* Remove existing CLASSGC support by @wasabii in https://github.com/ikvmnet/ikvm/pull/384
* RuntimeJavaType by @wasabii in https://github.com/ikvmnet/ikvm/pull/385
* Runtime Context by @wasabii in https://github.com/ikvmnet/ikvm/pull/393
* .NET 6.0 by @wasabii in https://github.com/ikvmnet/ikvm/pull/401
* OS X ARM64 by @wasabii in https://github.com/ikvmnet/ikvm/pull/402
* .NET 4.7.2 and .NET 6.0 by @wasabii in https://github.com/ikvmnet/ikvm/pull/406
* libnet and libnio by @wasabii in https://github.com/ikvmnet/ikvm/pull/409
* Rename win7-* packages and RIDs to win-*. .NET 8 no longer provides win7-* RIDs.
* IKVM.Image.runtime.* packages for OpenJDK native libraries.
* Use RAR for resolution of Framework references. Tools no longer package ref assemblies.
* libiava (libjava). Rename exists because Windows will not let a java.exe load a java.dll.
* libjvm. No external hosting support presently.
* Reintroduce libikvm for various platform specific functions that are difficult to P/Invoke.
* JNI loading logic more accurately behaves according to standard JDKs. IKVM.Image.runtime.* packages deliver native libraries, which are placed in the Image directory, not the .NET search path. Native libraries decorated with RPATH on Linux, and correctly mode rpath lookups on OS X.
* We lost AppDomain support due to the native libraries. The native OpenJDK libraries retain references to fields and classes in IKVM.Java. Loading two copies of IKVM.Java would use the same native libraries, with references to the fields from the wrong AppDomain. This will also cause issues for AssemblyLoadContext usage. As of now, avoid loading two copies of IKVM into the same process.
* New documentation structure, which should publish to ikvm.org.
* IKVM.Net.SDK should support MUSL.
* IKVM.Net.SDK now emits JNI header files by default.

## New Contributors
* @TheLastRar made their first contribution in https://github.com/ikvmnet/ikvm/pull/376

**Full Changelog**: https://github.com/ikvmnet/ikvm/compare/8.6.4...8.7.0-pre.1

<hr /><em>This discussion was created from the release <a href='https://github.com/ikvmnet/ikvm/releases/tag/8.7.0-pre.1'>8.7.0-pre.1</a>.</em>

**Comment by martin-honnen** [comment]
I have tried to use IKVM.NET.Sdk/8.7.0-pre.1 and IKVM to compile the existing Saxon HE .NET 10.9 code base (consisting of mostly Java code plus some .NET specific APIs) for .NET 4.8 and .NET 6.0. 
Building runs through without any errors and on trying to run code I get errors like
```
PS C:\Users\marti\source\repos\SaxonHE10Net\Query> .\bin\Release\net481\Query.exe -t -?

Unbehandelte Ausnahme: System.TypeInitializationException: Der Typeninitialisierer für "java.lang.Object" hat eine Ausnahme verursacht. ---> System.TypeInitializationException: Der Typeninitialisierer für "java.io.FileInputStream" hat eine Ausnahme verursacht. ---> System.TypeInitializationException: Der Typeninitialisierer für "IKVM.Runtime.LibJava" hat eine Ausnahme verursacht. ---> System.TypeInitializationException: Der Typeninitialisierer für "IKVM.Runtime.LibJvm" hat eine Ausnahme verursacht. ---> IKVM.Runtime.InternalException: Could not load libjvm.
   bei IKVM.Runtime.LibJvm..ctor()
   bei IKVM.Runtime.LibJvm..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei IKVM.Runtime.LibJava..ctor()
   bei IKVM.Runtime.LibJava..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei IKVM.Runtime.BootstrapClassLoader..ctor(RuntimeContext context)
   bei IKVM.Runtime.RuntimeClassLoaderFactory.GetBootstrapClassLoader()
   bei IKVM.Runtime.RuntimeClassLoaderFactory.GetClassLoaderWrapper(ClassLoader javaClassLoader)
   bei IKVM.Runtime.JNI.JNIFrame.GetFuncPtr(CallerID callerID, String clazz, String name, String sig)
   bei java.io.FileInputStream.initIDs()
   bei java.io.FileInputStream..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei java.io.FileInputStream.__<clinit>()
   bei java.lang.System.initializeSystemClass()
   bei IKVM.Runtime.JVM.EnsureInitialized()
   bei java.lang.Object..cctor()
   --- Ende der internen Ausnahmestapelüberwachung ---
   bei java.lang.Object..ctor()
   bei net.sf.saxon.Query..ctor() in Query.java:Zeile 49.
   bei Saxon.Cmd.DotNetQuery..ctor() in C:\Users\marti\source\repos\SaxonHE10Net\Query\Query.cs:Zeile 30.
   bei Saxon.Cmd.DotNetQuery.Main(String[] args) in C:\Users\marti\source\repos\SaxonHE10Net\Query\Query.cs:Zeile 40.
```

For .NET 6.0:
```
PS C:\Users\marti\source\repos\SaxonHE10Net\Query> .\bin\Release\net6.0\Query.exe -t -?
Unhandled exception. System.TypeInitializationException: The type initializer for 'java.lang.Object' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'java.io.FileInputStream' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'IKVM.Runtime.LibJava' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'IKVM.Runtime.LibJvm' threw an exception.
 ---> IKVM.Runtime.InternalException: Could not load libjvm.
   at IKVM.Runtime.LibJvm..ctor()
   at IKVM.Runtime.LibJvm..cctor()
   --- End of inner exception stack trace ---
   at IKVM.Runtime.LibJava..ctor()
   at IKVM.Runtime.LibJava..cctor()
   --- End of inner exception stack trace ---
   at IKVM.Runtime.BootstrapClassLoader..ctor(RuntimeContext context)
   at IKVM.Runtime.RuntimeClassLoaderFactory.GetBootstrapClassLoader()
   at IKVM.Runtime.RuntimeClassLoaderFactory.GetClassLoaderWrapper(ClassLoader javaClassLoader)
   at IKVM.Runtime.RuntimeClassLoader.FromCallerID(CallerID callerID)
   at IKVM.Runtime.JNI.JNIFrame.GetFuncPtr(CallerID callerID, String clazz, String name, String sig)
   at java.io.FileInputStream.initIDs()
   at java.io.FileInputStream..cctor()
   --- End of inner exception stack trace ---
   at java.io.FileInputStream.__<clinit>()
   at java.lang.System.initializeSystemClass()
   at __<MethodAccessor>__System__initializeSystemClass()
   at IKVM.Runtime.Accessors.Java.Lang.SystemAccessor.InvokeInitializeSystemClass()
   at IKVM.Runtime.JVM.EnsureInitialized()
   at java.lang.Object..cctor()
   --- End of inner exception stack trace ---
   at java.lang.Object..ctor()
   at net.sf.saxon.Query..ctor()
   at Saxon.Cmd.DotNetQuery..ctor() in C:\Users\marti\source\repos\SaxonHE10Net\Query\Query.cs:line 27
   at Saxon.Cmd.DotNetQuery.Main(String[] args) in C:\Users\marti\source\repos\SaxonHE10Net\Query\Query.cs:line 40
```

Are there any known changes (related perhaps to e.g. `System.out`, `System.err`, my guess as `java.lang.System.initializeSystemClass` appears in the stack trace)?
